### ISSUE TITLE
New Error Code with 0.11.1

### ISSUE THREAD

**Issue by JazzerAlto** [homebridge-xiaomi-roborock-vacuum/homebridge-xiaomi-roborock-vacuum#214]
I am on HB 1.1.0 and just upgraded to the 0.11.1 plugin.

My S6 threw an error and said it had a barrier in the way and couldn't return to the dock.  Looking at the log, I see a couple of interesting items:

This one after I docked from a room cleaning:

[5/23/2020, 09:17:07] [1st Floor Vacuum] ERR getState | {"code":"timeout"}

I am getting repeated errors in the log noted below.  Wonder if #193 is fixed or is this a new error code # that the plugin just doesn't know what to do with?  Restarting HB clears the error.

[5/23/2020, 09:44:55] [1st Floor Vacuum] WAR changedError | roborock.vacuum.s6 | Robot has an ERROR - 24, Unknown ERR | errorid can't be mapped. (24)

**Comment by netRunner0** [comment]
![IMG_0792](https://user-images.githubusercontent.com/44049295/82759070-2b325c80-9e04-11ea-9e60-b058c3a1a2f5.jpeg)

**Comment by diwiesign** [comment]
same here ...

```
[2020-5-25 7:41:11] [Staubsauger] INF ensureDevice | roborock.vacuum.s6 | The socket was destroyed or not initialised, initialising the device
[2020-5-25 7:41:11] [Staubsauger] ERR connect | miio.device, next try in 2 minutes | Error: Network communication is unavailable, device might be destroyed
[2020-5-25 7:41:11] [Staubsauger] ERR getState | {"device":null}
[2020-5-25 7:41:41] [Staubsauger] INF ensureDevice | roborock.vacuum.s6 | The socket was destroyed or not initialised, initialising the device
[2020-5-25 7:41:41] [Staubsauger] ERR connect | miio.device, next try in 2 minutes | Error: Network communication is unavailable, device might be destroyed
[2020-5-25 7:41:41] [Staubsauger] ERR getState | {"device":null}
[2020-5-25 7:42:11] [Staubsauger] INF ensureDevice | roborock.vacuum.s6 | The socket was destroyed or not initialised, initialising the device
[2020-5-25 7:42:11] [Staubsauger] ERR connect | miio.device, next try in 2 minutes | Error: Network communication is unavailable, device might be destroyed
[2020-5-25 7:42:11] [Staubsauger] ERR getState | {"device":null}
[2020-5-25 7:42:41] [Staubsauger] INF ensureDevice | roborock.vacuum.s6 | The socket was destroyed or not initialised, initialising the device
[2020-5-25 7:42:41] [Staubsauger] ERR connect | miio.device, next try in 2 minutes | Error: Network communication is unavailable, device might be destroyed
[2020-5-25 7:42:41] [Staubsauger] ERR getState | {"device":null}
[2020-5-25 7:43:11] [Staubsauger] INF ensureDevice | roborock.vacuum.s6 | The socket was destroyed or not initialised, initialising the device
[2020-5-25 7:43:11] [Staubsauger] ERR connect | miio.device, next try in 2 minutes | Error: Network communication is unavailable, device might be destroyed
[2020-5-25 7:43:11] [Staubsauger] ERR getState | {"device":null}
[2020-5-25 7:43:41] [Staubsauger] INF ensureDevice | roborock.vacuum.s6 | The socket was destroyed or not initialised, initialising the device
[2020-5-25 7:43:41] [Staubsauger] ERR connect | miio.device, next try in 2 minutes | Error: Network communication is unavailable, device might be destroyed
[2020-5-25 7:43:41] [Staubsauger] ERR getState | {"device":null}
[2020-5-25 7:44:11] [Staubsauger] INF ensureDevice | roborock.vacuum.s6 | The socket was destroyed or not initialised, initialising the device
[2020-5-25 7:44:11] [Staubsauger] ERR connect | miio.device, next try in 2 minutes | Error: Network communication is unavailable, device might be destroyed
[2020-5-25 7:44:11] [Staubsauger] ERR getState | {"device":null}
[2020-5-25 7:44:41] [Staubsauger] INF ensureDevice | roborock.vacuum.s6 | The socket was destroyed or not initialised, initialising the device
[2020-5-25 7:44:41] [Staubsauger] ERR connect | miio.device, next try in 2 minutes | Error: Network communication is unavailable, device might be destroyed
[2020-5-25 7:44:41] [Staubsauger] ERR getState | {"device":null}
[2020-5-25 7:45:11] [Staubsauger] INF ensureDevice | roborock.vacuum.s6 | The socket was destroyed or not initialised, initialising the device
[2020-5-25 7:45:11] [Staubsauger] ERR connect | miio.device, next try in 2 minutes | Error: Network communication is unavailable, device might be destroyed
[2020-5-25 7:45:11] [Staubsauger] ERR getState | {"device":null}
```

**Comment by diwiesign** [comment]
Unfortunately I can't go back to version 0.8.0. After uninstalling the plug-in, changing the configuration and restarting the Raspi, version 0.8.0 was installed according to the instructions with the following error message.

Anyone have an idea how I can get back to a working version? Thank you so much!

```
[2020-5-25 13:12:16] [Staubsauger] ERR getCleaning | Failed getting the cleaning status. TypeError: Cannot read property 'property' of null
    at XiaomiRoborockVacuum.get isCleaning [as isCleaning] (/usr/local/lib/node_modules/homebridge-xiaomi-roborock-vacuum/index.js:763:32)
    at XiaomiRoborockVacuum.getCleaning (/usr/local/lib/node_modules/homebridge-xiaomi-roborock-vacuum/index.js:769:31)
    at /usr/local/lib/node_modules/homebridge-xiaomi-roborock-vacuum/index.js:147:49
    at module.exports (/usr/local/lib/node_modules/homebridge-xiaomi-roborock-vacuum/lib/callbackify.js:5:26)
    at On.<anonymous> (/usr/local/lib/node_modules/homebridge-xiaomi-roborock-vacuum/index.js:147:26)
    at On.emit (events.js:310:20)
    at On.EventEmitter.emit (/usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/src/lib/EventEmitter.ts:42:22)
    at On.Characteristic._this.getValue (/usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/src/lib/Characteristic.ts:462:12)
    at /usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/src/lib/Accessory.ts:1215:22
    at Array.forEach (<anonymous>)
    at Bridge.Accessory._this._handleGetCharacteristics (/usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/src/lib/Accessory.ts:1143:10)
    at HAPServer.emit (events.js:310:20)
    at HAPServer.EventEmitter.emit (/usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/src/lib/EventEmitter.ts:42:22)
    at HAPServer._this._handleCharacteristics (/usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/src/lib/HAPServer.ts:756:12)
    at IncomingMessage.<anonymous> (/usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/src/lib/HAPServer.ts:280:24)
    at IncomingMessage.emit (events.js:310:20)
    at endReadableNT (_stream_readable.js:1187:12)
    at processTicksAndRejections (internal/process/task_queues.js:84:21)
[2020-5-25 13:12:16] [Staubsauger] ERR getPauseState | No vacuum cleaner is discovered yet.
```

**Comment by afharo** [comment]
It looks to me like the error is about failing to connect to the robot.

I'm running the latest for my `roborock.vacuum.s5` and it's connecting fine. I can see you all reported the error with the S6 model.

Any other details you can provide?
- Has that model's firmware been recently updated? 
- What version of `node` are you running it with?
- Any special configuration you are using? `delay: true`

Re the downgrade to `0.8.0`, it should be pretty straightforward. Nothing crazy on our end. The error you show from that version confirms the same issue: can't connect so can't read the `isCleaning` property.

**Comment by diwiesign** [comment]
Thank you @afharo 

> I'm running the latest for my `roborock.vacuum.s5` and it's connecting fine. I can see you all reported the error with the S6 model.

yes, S6 model

> Any other details you can provide?
> 
> * Has that model's firmware been recently updated?

No

> * What version of `node` are you running it with?

v12.16.3

> * Any special configuration you are using? `delay: true`

no
```
        {
            "name": "Staubsauger",
            "ip": "192.168.xx.xx",
            "token": "xxxxxxxxxxxxxxxxxxxxxx",
            "waterBox": false,
            "pause": true,
            "dock": true,
            "delay": false,
            "cleanword": "Sauge",
            "autoroom": false,
            "rooms": [
                {
                    "name": "Flur",
                    "id": 16
                },
                {
                    "name": "Wohnzimmer",
                    "id": 17
                },
                {
                    "name": "Küche",
                    "id": 18
                },
                {
                    "name": "Kinderzimmer",
                    "id": 19
                },
                {
                    "name": "Badezimmer",
                    "id": 20
                }
            ],
            "accessory": "XiaomiRoborockVacuum"
        }
```

> 
> Re the downgrade to `0.8.0`, it should be pretty straightforward. Nothing crazy on our end. The error you show from that version confirms the same issue: can't connect so can't read the `isCleaning` property.

For me downgrade doesn't works and I had to recover a complete Pi Backup.

**Comment by JazzerAlto** [comment]
The issue I reported for me is not the same as what the other replies contain in this thread.  I am seeing an occasional err: [5/26/2020, 03:55:18] [1st Floor Vacuum] ERR getState | {"code":"timeout"}

and I also received the error below when my S6 had an error condition which was caused by getting stuck in a blocked area.  I had to move the S6 and start up the cleaning again but even after that, this error code continued to repeated show in the logs until I rebooted HB:

[5/23/2020, 09:44:55] [1st Floor Vacuum] WAR changedError | roborock.vacuum.s6 | Robot has an ERROR - 24, Unknown ERR | errorid can't be mapped. (24)

I am not seeing any of the socket errors other have replied with.

Both my S6s appear to still be working fine.

**Comment by afharo** [comment]
oh! Good to know!
So code 24 is `Stuck in a blocked area`!

Re the repeating error it might be related to #162

**Comment by afharo** [comment]
The repeating error should be fixed in `homebridge-xiaomi-roborock-vacuum@0.11.4`.

Closing this issue. Please open a new one about the S6 connection problem if it still occurs.

**Comment by JazzerAlto** [comment]
Thanks @afharo!  I'll keep an eye out and advise if I see any mores issues.
